File: selection_sort.js
const arr = [10, 5, 2, 8, 7];
function selection_sort(arr) {
    for (let i = 0; i < arr.length; i++) {
        let small = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[small]) {
                small = j;
            }
        }
        if (i !== small) {
            let temp = arr[i];
            arr[i] = arr[small];
            arr[small] = temp;
        }
    }
}
selection_sort(arr);
console.log(arr);

File: selection_sort.c
#include <stdio.h>
void selection_sort();
int main()
{
    int array[5] = {3, 2, 4, 1, 5};
    selection_sort(array, 5);

    return 0;
}

void selection_sort(int *arr[], int len)
{
    int i, j, small;
    for (i = 0; i < len; i++)
    {
        small = i;
        for (j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[small])
            {
                small = j;
            }
        }
        if (i != small)
        {
            int temp = arr[i];
            arr[i] = arr[small];
            arr[small] = temp;
        }
    }
}